connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -108,18 +107,15 @@ public Map<String, ByteBuffer> performAssignment(String leaderId, String protoco
         log.debug("Max config offset root: {}, local snapshot config offsets root: {}",
                   maxOffset, coordinator.configSnapshot().offset());
 
-        short protocolVersion = memberConfigs.values().stream()
-            .allMatch(state -> state.assignment().version() == CONNECT_PROTOCOL_V2)
-                ? CONNECT_PROTOCOL_V2
-                : CONNECT_PROTOCOL_V1;
+        short protocolVersion = ConnectProtocolCompatibility.fromProtocol(protocol).protocolVersion();
```

Comment:
I think this change makes sense because the `protocol` is decided by group coordinator, which is selected by one protocol supported by all clients. So deciding the protocol version by `protocol` is fine. Thanks.

Reply:
Yep, exactly 👍
Should've known that when I implemented KIP-507 originally but was still getting my bearings with the group coordinator logic.